translate/translate.go

```diff
@@ -18,28 +18,49 @@ type Content struct {
 	Slug        string
 }
 
-func (tc *TranslationContext) Translate(item mapper.Item, itemType mapper.Type) (fileName string, content string) {
-
-	rawContent := tc.convertContent(item.Fields, itemType.Fields)
-	if tc.TransConfig.Encoding == "yaml" {
-		content = rawContent.ToYaml()
-	} else {
-		content = rawContent.ToToml()
+// UnionValuesAndOverride takes a defaults and an overrides map and assigns any missing
+// values from the defaults to the overrides map.
+func (tc *TranslationContext) UnionValuesAndOverride(itemDefault map[string]interface{}, itemOverride map[string]interface{}) (combinedItem map[string]interface{}) {
+	for k, v := range itemDefault {
+		if itemOverride[k] == nil {
+			itemOverride[k] = v
+		} else if s, ok := itemOverride[k].(string); ok && s == "" {
+			itemOverride[k] = v
+		} else if s, ok := itemOverride[k].([]string); ok && len(s) == 0 {
+			itemOverride[k] = v
+		} else if s, ok := itemOverride[k].([]interface{}); ok && len(s) == 0 {
+			itemOverride[k] = v
+		}
```

Comment:
what about `int`, `bool` or other types that I can't think of? Try to explore an approach to check for nil or zero value like the one [here] (https://stackoverflow.com/a/13906031/1563178) so that we can (hopefully) support all types in one shot